### ISSUE TITLE
[TTNN] Expanded RoPE fusing pattern produces incorrect concat at batch=1, corrupting decode output

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-mlir#8042]
### Problem description

The RoPE fusing pattern introduced in #7751 (`16ec8f80f`) generates incorrect `ttnn.concat` ops in the decode graph when `batch=1` and `optimization_level=1`. The result is garbage output from Llama models at batch=1 while batch=2 produces correct output.

Reported in tt-xla: https://github.com/tenstorrent/tt-xla/issues/4325

### Bisect result

- **First bad commit**: `16ec8f80fbc34da0fab3fd847d94aca240b14082` — `[TTNN] Add expanded (trig-identity) RoPE fusing pattern (#7751)`
- **Last good commit**: `c5575fc46d70ee245a36d69488e0a3781eb269a0` — `[EmitPy] Refactor EmitPy CPU-hoisting (#7706)`

### Root cause theory

In the bad commit's decode graph, the RoPE concat ops are passed the same tensor twice instead of the two complementary halves:

```
// Bad (batch=1, opt_level=1)
%26 = "ttnn.concat"(%23, %23) <{dim = 3 : si32}>   // sin concat'd with itself
%27 = "ttnn.concat"(%25, %25) <{dim = 3 : si32}>   // cos concat'd with itself
```

```
// Good (last good commit)
%34 = "ttnn.concat"(%30, %33) <{dim = 3 : si32}>   // two distinct tensors
```

The fusing pattern appears to incorrectly CSE or alias the two RoPE half-arguments at batch=1, causing both inputs to the concat to resolve to the same value. This corrupts the rotary embeddings for every transformer layer in the decode step.

Prefill is unaffected (prefill logits are identical between good and bad commits). Only the decode forward pass at batch=1 is broken.

### Conditions

- **Model**: Llama-3.2-3B (and likely other Llama/RoPE-based models)
- **Batch size**: 1 only — batch=2 produces correct output
- **optimization_level**: 1 only — opt_level=0 is correct

### Decode graph IR (before/after bad commit)

https://gist.github.com/kmabeeTT/a6f8b027d1d87e24749c0d7642cb1441

Contains `graph_3_ttir.mlir` and `graph_3_ttnn.mlir` for both the good and bad commits. The relevant concat ops are around line 301 in the bad ttnn graph.

**Comment by sdjordjevicTT** [comment]
Hi @kmabeeTT,

@djstefanovicTT managed to fix this issue, but we are not sure what to run to validate the PCC. Can you try PR change, or point us to the exact tt-xla command to verify the fix?

**Comment by kmabeeTT** [comment]
Thank you @djstefanovicTT @sdjordjevicTT! Whoops, apologies didn't include command to check in tt-mlir - was super quick and easy for me to try in tt-xla, confirmed this fix cherry picked works to solve the bug... thumbs up!  


Before fix:

```
prompt: Tell me a story., output:  Tell in contexto companies of continuing late protective variations of evil tofulness corn or divorce & buddy assembly, operational means museums type asset for notice than and doors especially how into thatiele and size doesn cells popular loss when remedies-four such profiles about, the recommend use of critical.
, you he press tried eliminated and: the work
PASSED
```

After Fix:
```
prompt: Tell me a story., output:  For me, stories have always provided a means to make sense of the world. Storytelling is a skill I’ve honed, though it’s far from perfect. My wife would tell you that my jokes are awful.
My storytelling professor in college was a man of great influence. His name was John W. Bradley
PASSED
```
